pkg/controller/discovery.go

```diff
@@ -22,6 +22,7 @@ func (c *CoastguardController) OnAdd(clusterID string, kubeConfig *rest.Config)
 
 func (c *CoastguardController) addCluster(clusterID string, clientSet kubernetes.Interface) {
 	rc := remotecluster.New(clusterID, clientSet)
+	rc.SetEventChannel(c.clusterEvents)
```

Comment:
Remove the SetEventChannel functionality and create new remote cluster object directly with the clusterEvents channel.

This will avoid the need for locking/etc inside remote cluster..